### PR TITLE
feat: split config into node and provider configs

### PR DESCRIPTION
**Pull request by zarazan**
## Description

- Added an optional "config_dir" parameter to the price-feeder config that will optionally load a folder of toml files for provider configs.
- Added a sample `ojo-provider-config` directory

closes: #124

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -211,19 +214,24 @@ func (c Config) ExpectedSymbols() []string {
...
 // An error is returned if reading or parsing the config fails.
 func ParseConfig(configPath string) (Config, error) {
-	var cfg Config
+	return ParseConfigs([]string{configPath})
+}
 
-	if configPath == "" {
-		return cfg, ErrEmptyConfigPath
-	}
+func ParseConfigs(configPaths []string) (Config, error) {
```
Comment (adamewozniak) [inline config/config.go:220]:
```suggestion
// ParseConfigs attempts to read and parse configuration from the given file paths.
// An error is returned if reading or parsing the config fails.
func ParseConfigs(configPaths []string) (Config, error) {
```

File: config/config.go [hunk 2]
```diff
@@ -211,19 +214,24 @@ func (c Config) ExpectedSymbols() []string {
...
-	if err := viper.ReadInConfig(); err != nil {
-		return cfg, fmt.Errorf("failed to read config: %w", err)
+	for _, configPath := range configPaths {
+		if configPath == "" {
+			return cfg, ErrEmptyConfigPath
+		}
+		viper.SetConfigFile(configPath)
+		if err := viper.MergeInConfig(); err != nil {
+			return cfg, fmt.Errorf("failed to read config: %w", err)
+		}
```
[the remark this answers is not in the document]
Comment (adamewozniak) [reply config/config.go:234]: Maybe add a comment here saying that we expect two types of files?

File: cmd/price-feeder.go [hunk 3]
```diff
@@ -61,6 +62,7 @@ func init() {
 	rootCmd.PersistentFlags().String(flagLogLevel, zerolog.InfoLevel.String(), "logging level")
 	rootCmd.PersistentFlags().String(flagLogFormat, logLevelText, "logging format; must be either json or text")
 	rootCmd.PersistentFlags().Bool(flagSkipProviderCheck, false, "skip the coingecko API provider check")
+	rootCmd.PersistentFlags().String(flagProviderConfig, config.DefaultProviderConfigPath, "provider config file path")
```
[the remark this answers is not in the document]
Comment (zarazan) [reply cmd/price-feeder.go:65]: I prefer that as well but am worried about it being a breaking change (could break some people's devops scripts).

**Comment by adamewozniak** [comment]
Btw, let's wait to backport this into the umee branch until we've cut an 0.47-compatible price feeder for them

**Comment by adamewozniak** [comment]
> Btw, let's wait to backport this into the umee branch until we've cut an 0.47-compatible price feeder for them

Never mind, in standup we decided to wait on merging the 0.47 changes to umee

**Comment by zarazan** [comment]
Updated to use a folder for provider configs based on slack discussion. `Maybe the provider config directory is an environment variable? `

File: README.md [hunk 4]
```diff
@@ -41,23 +41,28 @@ The list of current supported providers:
...
 - [Okx](https://www.okx.com/)
-- [Osmosis](https://app.osmosis.zone/)
 - [OsmosisV2](https://github.com/ojo-network/osmosis-api)
+- [Polygon](https://api.polygon.io)
 <!-- markdown-link-check-enable -->
 
 ## Usage
 
-The `price-feeder` tool runs off of a single configuration file. This configuration
+The `price-feeder` tool runs off of two configuration files. The provider-config
```
Comment (rbajollari) [inline README.md:53]: Could be more than two files now that the provider-config is a directory

File: README.md [hunk 5]
```diff
@@ -41,23 +41,28 @@ The list of current supported providers:
...
 file defines what exchange rates to fetch and what providers to get them from.
-In addition, it defines the oracle's keyring and feeder account information.
+The node-config defines the oracle's keyring and feeder account information.
 The keyring's password is defined via environment variables or user input.
 More information on the keyring can be found [here](#keyring)
 Please see the [example configuration](price-feeder.example.toml) for more details.
 
+The default `provider_config.toml` is used when no --provider-config flag is provided.
```
Comment (rbajollari) [inline README.md:60]:
```suggestion
The default provider configs in the `provider-config` directory are used when no --provider-config flag is provided.
```

**Review by rbajollari — APPROVED** [review]
Looks good! README just needs to be updated now that provider configs are in its own directory.

File: README.md (hunk 4 above)
Comment (zarazan) [reply README.md]: Good call, README has been updated with some clarification.

**Comment by zarazan** [comment]
PR has been updated based on community feedback. The way the price-feeder is ran from the command line will stay the same with one required argument - the node config file. You can optionally set the "config_dir" inside that file to load a folder of toml files that set the currency_pairs, deviation_thresholds, and provider_endpoints.

**Review by adamewozniak — APPROVED** [review]
Re-approving to ack the changes 👍

**Pull request merged**